tools/build-bpf-skel.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1+
+
+import argparse
+import logging
+import pathlib
+import shutil
+import subprocess
+import sys
+import tempfile
+
+def clang_arch_flag(arch):
+    return '-D__{}__'.format(arch)
+
+
+def target_triplet():
+    gcc_exec = None
+    try:
+        gcc_exec = shutil.which('gcc')
+    except shutil.Error as e:
+        logging.error('Failed to get which gcc: {}'.format(e))
+        return None
+
+    try:
+        return subprocess.check_output([gcc_exec, '-dumpmachine'],
+                text=True).strip()
+    except subprocess.CalledProcessError as e:
+        logging.error('Failed to get target triplet: {}'.format(e))
+        return None
+
+
+def clang_compile(clang_exec, clang_flags, src_c, out_file, target_arch,
+        target_triplet):
+    clang_args = [clang_exec]
+    clang_args.extend(clang_flags)
+    clang_args.extend([target_arch])
+
+    clang_args.extend([
+        '-I.'])
+
+    if target_triplet:
+        clang_args.extend([
+            '-isystem',
+            '/usr/include/{}'.format(target_triplet)])
+
+    clang_args.extend([
+        '-idirafter',
+        '/usr/local/include',
+        '-idirafter',
+        '/usr/include'])
+
+    clang_args.extend([src_c])
+    clang_args.extend(['-o', out_file])
+
+    logging.debug('Generating LLVM bitcode *.bc:')
```

Comment:
log message is wrong, we are not generating intermediate bitcode, we go straight to ELF object file with `-target bpf -o <file>`